### PR TITLE
Comparison IoTDB with other TSDBs

### PR DESCRIPTION
**Pull request by jixuan1989**
Hi, I write a doc for comparing IoTDB with other TSDBs:
- InfluxDB, 
- OpenTSDB, 
- KairosDB, and 
- TimeSacleDB.

The article includes:
- Feature comparison;
- Performance comparison.

The performance experiments do not follow the latest version, which needs more effort to fix it in the future.

### CODE REVIEW

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+| Value filter                | o                           | o          | x          | x          | o           |   
+| Downsampling                | **O**                       | o          | o          | o          | o           |   
+| Fill                        | **O**                       | o          | o          | x          | o           |   
+| LIMIT                       | o                           | o          | o          | o          | o           |   
+| SLIMIT                      | o                           | o          | x          | x          | ?           |   
+| Latest value                | O                           | o          | o          | x          | o           |
+
+**Details**
+
+* OpenSOurce:
```
Comment (liutaohua) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:91]: OpenSource

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 2]
```diff
@@ -0,0 +1,400 @@
...
+| SLIMIT                      | o                           | o          | x          | x          | ?           |   
+| Latest value                | O                           | o          | o          | x          | o           |
+
+**Details**
+
+* OpenSOurce:  
+ 
+  * IoTDB uses Apache License 2.0 and it is in Apache incubator. 
+  * InfluxDB uses MIT license. However, **the cluster version is not open sourced**.
+  * OpenTSDB uses LGPL2.1, which **is not compatible with Apache License**.
```
Comment (liutaohua) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:95]: How about describing what's special about the LGPL?
eg:
This means that any changes you make must also be open source

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 3]
```diff
@@ -0,0 +1,400 @@
...
+  * KairosDB uses Apache License 2.0.
+  * TimescaleDB uses Timescale License, which is not free for enterprise. 
+
+* SQL like: 
+
+  * IoTDB and InfluxDB supports SQL like language. Besides, The integration of IoTDB and Calcite is alomost done (a PR has been submitted), which means IoTDB will support Standard SQL.
```
Comment (liutaohua) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:101]: I think it's better to spell out the future version here than to say what we'll support in the future.
eg:
The integration of IoTDB and Calcite will in 0.10.1

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 4]
```diff
@@ -0,0 +1,400 @@
...
+| LIMIT                       | o                           | o          | o          | o          | o           |   
+| SLIMIT                      | o                           | o          | x          | x          | ?           |   
+| Latest value                | O                           | o          | o          | x          | o           |
+
+**Details**
+
+* OpenSOurce:  
+ 
+  * IoTDB uses Apache License 2.0 and it is in Apache incubator. 
+  * InfluxDB uses MIT license. However, **the cluster version is not open sourced**.
```
Comment (vesense) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:94]: How about adding a separate section/item to highlight distributed functions?
This is an important advantage of iotdb over others.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 5]
```diff
@@ -0,0 +1,400 @@
...
+|-----------------------------|---------------------------------|------------|------------|------------|-------------|   
+| Align by time               | **O**                           | o          | x          | x          | o           |   
+| Compression                 | **O**                           | :\-(       | :\-\(      | :\-\(      | :\-\(       |   
+| MQTT support                | **O**                           | o          | x          | x          | :\-\(       |   
+| Run on Edge-side Device     | **O**                           | o          | x          | :\-\(      | o           |   
+| Multi\-instance Sync        | **O**                           | x          | x          | x          | x           |   
+| JDBC Driver                 | **o**                           | x          | x          | x          | x           |   
+| Standard SQL                | o                               | x          | x          | x          | **O**       |   
+| Spark integration           | **O**                           | x          | x          | x          | x           |
```
Comment (vesense) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:222]: add flink integration

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 6]
```diff
@@ -0,0 +1,400 @@
...
+  Interface: InfluxQL and HTTP API
+
+* OpenTSDB and KairosDB - Time series database based on NoSQL
+
+  These two DBs are similar, while the first is based on HBase and the second is based on Cassandra.
+  Both of them provides RESTful style API.
+  
+  Interface: Restful API
+
+* TimeSacleDB - Time series database based on Relational Database
```
Comment (Alima777) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:44]: TimescaleDB

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md (hunk 3 above)
Comment (Alima777) [reply docs/UserGuide/9-Comparison/TSDB-Comparison.md:101]: I agree with @liutaohua here. And a little typo: almost.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 7]
```diff
@@ -0,0 +1,400 @@
...
+
+  * IoTDB: IoTDB proposes a [Tree based schema](http://iotdb.apache.org/UserGuide/Master/2-Concept/1-Data%20Model%20and%20Terminology.html#data-model-and-terminology). 
+   It is quite different with other TSDBs. However, the kind of schema has the following advantages:
+    
+    * In many industrial scenarios, the management of devices are hierarchical, rather than flat.
+    That is why we think a tree based schema is better than tag-value based schema.
+    
+    * In many real world applications, tag names are constant. For example, a wind turbine manufacturer
+    always identify their wind turbines by which country it locates, the farm name it belongs to, and it ID in the farm.
+    So, a 4-depth tree ("root.the-country.the-farm.the-id") is fine.
```
Comment (Alima777) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:115]: typo: its ID. And. maybe "root.countryName.farmName.id" is better I think.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 8]
```diff
@@ -0,0 +1,400 @@
...
+    
+    * In many real world applications, tag names are constant. For example, a wind turbine manufacturer
+    always identify their wind turbines by which country it locates, the farm name it belongs to, and it ID in the farm.
+    So, a 4-depth tree ("root.the-country.the-farm.the-id") is fine. 
+    You do not need to repeat to tell IoTDB the 2nd level of the tree is for country name, 
+    the 3rd level is for farm id, etc..
+    
+    * A path based time series ID definition also supports flexible queries, like "root.\*.a.b.\*", wehre \* is wildcard character.
+  
+  * InfluxDB, KairosDB, OpenTSDB are tag-value based, which is more popular currently.
```
Comment (Alima777) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:121]: How about putting this before IoTDB, to highlight the advantage of  tree based schema.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 9]
```diff
@@ -0,0 +1,400 @@
...
+  
+  * TimescaleDB uses relational table.   
+
+* Order by time:
+  
+  Order by time seems quite trivil for time series database. But... if we consider another featuer, called align by time,
```
Comment (Alima777) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:127]: typo: feature.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 10]
```diff
@@ -0,0 +1,400 @@
...
+  e.g., a SQL returns 1 point per 5 minutes and start with 2020-04-27 08:00:00 while another SQL returns 1 point per 5 minutes + 10 seconds and start with 2020-04-27 08:00:01.
+  （InfluxDB also supports adhoc downsampling but the performance is ..... hm)
+  
+  * There is no disk loss for IoTDB.
+  
+
+* Fill:
+
+  Sometimes we thought the data is collected in some fixed frequency, e.g., 1Hz (1 point per second). 
+  But usually, we may lost some data points, because the network is unstalbe, the machine is busy, or the machine is down for several minutes.
```
Comment (Alima777) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:179]: typo: unstable.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 11]
```diff
@@ -0,0 +1,400 @@
...
+  Sharing nothing architecture is good, but sometimes you have to add new servers even your CPU and memory is idle but the disk is full...
+  Besides, if we can save the data file directly to HDFS, it will be more easy to use Spark and other softwares to analyze data, without ETL.
+  
+  * IoTDB supports write data locally or on HDFS directly. IoTDB also allows user extend to store data on other NFS.
+  * InfluxDB, KairosDB have to write data locally.
+  * OpenTSDB has to write data on HDFS.
+    
+**Conclusion**:    
+  
+  We can find that IoTDB has many interesting features that other TSDBs do not support.
```
Comment (Alima777) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:275]: I think "interesting" is not suitable here, which seems to be not necessary. If i am a user of iotdb, what i want to see in the conclusion is " IoTDB has more useful and practical function and features than other TSDBs".

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md (hunk 1 above)
Comment (qiaojialin) [reply docs/UserGuide/9-Comparison/TSDB-Comparison.md]: fixed

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md (hunk 5 above)
Comment (qiaojialin) [reply docs/UserGuide/9-Comparison/TSDB-Comparison.md:222]: fixed

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 12]
```diff
@@ -0,0 +1,384 @@
...
+| Hive integration            | **O**                           | x          | x          | x          | x           |
+| Writing data to NFS (HDFS)  | **O**                           | x          | o          | x          | x           |
+| Flink integration           | **O**                           | x          | x          | x          | x           |
+
+
+* Align by time: have been introduced. Let's skip it..
+
+* Compression: 
+  * IoTDB supports many encoding and compression for time series, like RLE, 2DIFF, Gorilla, etc.. and Snappy compression.
+  In IoTDB, you can choose which encoding method you want, according to the data distribution. For more info, see [here](http://iotdb.apache.org/UserGuide/Master/2-Concept/3-Encoding.html).
```
Comment (HTHou) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:232]: Please update the URL... I think we have removed the serial num in doc file names. See [PR#1256](https://github.com/apache/incubator-iotdb/pull/1256).

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md [hunk 13]
```diff
@@ -0,0 +1,384 @@
...
+
+* SQL like: 
+
+  * IoTDB and InfluxDB supports SQL like language. Besides, The integration of IoTDB and Calcite is alomost done (a PR has been submitted), which means IoTDB will support Standard SQL.
+  * OpenTSDB and KairosDB only support Rest API. Besides, IoTDB also supports Rest API (a PR has been submitted).
+  * TimescaleDB uses the SQL the same with PG.
+  
+* Schema:
+
+  * IoTDB: IoTDB proposes a [Tree based schema](http://iotdb.apache.org/UserGuide/Master/2-Concept/1-Data%20Model%20and%20Terminology.html#data-model-and-terminology).
```
Comment (HTHou) [inline docs/UserGuide/9-Comparison/TSDB-Comparison.md:107]: Please update the URL too.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md (hunk 4 above)
Comment (jixuan1989) [reply docs/UserGuide/9-Comparison/TSDB-Comparison.md:94]: We can do that after our cluster mode is published.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md (hunk 3 above)
Comment (jixuan1989) [reply docs/UserGuide/9-Comparison/TSDB-Comparison.md]: will do this after the PR is merged.

File: docs/UserGuide/9-Comparison/TSDB-Comparison.md (hunk 2 above)
Comment (jixuan1989) [reply docs/UserGuide/9-Comparison/TSDB-Comparison.md]: It is hard to say... LGPL is more loose that GPL. It allows users do not open their codes in some conditions.

**Pull request merged**